src/payment/bolt11.rs

```diff
@@ -469,41 +463,18 @@ impl Bolt11Payment {
 		&self, amount_msat: Option<u64>, description: &str, expiry_secs: u32,
 		manual_claim_payment_hash: Option<PaymentHash>,
 	) -> Result<Bolt11Invoice, Error> {
-		let currency = Currency::from(self.config.network);
-		let keys_manager = Arc::clone(&self.keys_manager);
-		let duration = SystemTime::now()
-			.duration_since(SystemTime::UNIX_EPOCH)
-			.expect("for the foreseeable future this shouldn't happen");
+		let invoice_description = Bolt11InvoiceDescription::Direct(
+			Description::new(description.to_string()).map_err(|_| Error::InvoiceCreationFailed)?,
+		);
 
 		let invoice = {
-			let invoice_res = if let Some(payment_hash) = manual_claim_payment_hash {
-				create_invoice_from_channelmanager_and_duration_since_epoch_with_payment_hash(
-					&self.channel_manager,
-					keys_manager,
-					Arc::clone(&self.logger),
-					currency,
-					amount_msat,
-					description.to_string(),
-					duration,
-					expiry_secs,
-					payment_hash,
-					None,
-				)
-			} else {
-				create_invoice_from_channelmanager_and_duration_since_epoch(
-					&self.channel_manager,
-					keys_manager,
-					Arc::clone(&self.logger),
-					currency,
-					amount_msat,
-					description.to_string(),
-					duration,
-					expiry_secs,
-					None,
-				)
-			};
-
-			match invoice_res {
+			let mut invoice_params = Bolt11InvoiceParameters::default();
+			invoice_params.amount_msats = amount_msat;
+			invoice_params.description = invoice_description;
+			invoice_params.invoice_expiry_delta_secs = Some(expiry_secs);
+			invoice_params.payment_hash = manual_claim_payment_hash;
```

Comment:
Consider using the `..Default::default()` struct initialization syntax to avoid repeating `invoice_params`.